src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/CompletionListMerger.cs

```diff
@@ -141,22 +141,22 @@ private static void TrySplitJsonElement(object data, List<JObject> collector)
         }
         else
         {
-            collector.Add((JObject)JsonHelpers.TryConvertFromJsonElement(jsonElement).AssumeNotNull());
+            collector.Add(jsonElement);
         }
     }
 
-    private static void TrySplitJObject(object data, List<JObject> collector)
+    private static void TrySplitJObject(object data, List<JsonElement> collector)
     {
-        if (data is not JObject jobject)
+        if (data is not JObject jObject)
```

Comment:
There is another method that checks for JsonElement. This comes from https://github.com/dotnet/razor/pull/10494 and making sure Razor can accept anything from delegated servers, allowing them to move engines without insertion annoyances. Similar to https://github.com/dotnet/razor/pull/10551#discussion_r1657873223 the changes here are because the result needs to match what the Razor LSP server users, but maintain that agnosticism in what it accepts. And I fixed the casing of a variable :)

Reply:
Got it, I'm seeing why now. Also thank you for fixing the casing 🥇